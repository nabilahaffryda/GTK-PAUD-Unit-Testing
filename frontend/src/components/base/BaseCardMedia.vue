<template>
  <v-list color="transparent">
    <v-list-item :href="url">
      <v-list-item-content class="pa-0 ma-0">
        <v-card flat :class="['align-center text-center', { 'py-4': !noPadding }]" :color="color">
          <v-card-text :class="{ 'pa-0': noPadding }">
            <v-avatar :size="size" :color="avatarColor" :tile="tile">
              <slot />
            </v-avatar>
          </v-card-text>
        </v-card>
      </v-list-item-content>
    </v-list-item>
    <v-list-item class="text-center">
      <v-list-item-content :class="{ 'pa-0': noPadding, 'py-1': !noPadding }">
        <v-list-item-title class="font-weight-bold title" v-html="title" />
        <p class="body-2 grey--text" v-html="desc" />
        <p class="caption font-weight-bold my-2 align-bottom">
          <slot name="status" />
        </p>
      </v-list-item-content>
    </v-list-item>
  </v-list>
</template>

<script>
export default {
  name: 'CardMedia',
  props: {
    noPadding: {
      type: Boolean,
      default: false,
    },
    tile: {
      type: Boolean,
      default: false,
    },
    avatarColor: {
      type: String,
      default: 'yellow lighten-3',
    },
    color: {
      type: String,
      default: '',
    },
    size: {
      type: Number,
      default: 95,
    },
    title: {
      type: String,
      default: '',
    },
    desc: {
      type: String,
      default: '',
    },
    url: {
      type: String,
      default: '',
    },
  },
};
</script>

<style scoped></style>
