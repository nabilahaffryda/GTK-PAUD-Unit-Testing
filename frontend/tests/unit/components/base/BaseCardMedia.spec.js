import { mount, createLocalVue } from "@vue/test-utils";
import BaseCardMedia from "@/components/base/BaseCardMedia.vue";
import Vuetify from 'vuetify';
import Vue from 'vue';

Vue.use(Vuetify)
const localVue = createLocalVue();

describe('BaseCardMedia.vue', () => {
    let vuetify
    beforeEach(() => {
        vuetify = new Vuetify()
    })
    test('should show card media', () => {
        const VCard = {
            props: ['noPadding', 'tile', 'avatarColor', 'color', 'size', 'title', 'desc', 'url'],
            template: '<div><slot name="status" /></div>'
        }

        const wrapper = mount(BaseCardMedia, {
            localVue,
            vuetify,
            stubs: {
                VCard
            }
        });
        expect(wrapper).toMatchSnapshot();
    })
})
