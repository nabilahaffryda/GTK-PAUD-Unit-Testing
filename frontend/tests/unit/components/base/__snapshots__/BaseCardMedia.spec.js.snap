// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseCardMedia.vue should show card media 1`] = `
<div role="list" class="v-list v-sheet theme--light transparent">
  <div tabindex="-1" role="listitem" class="v-list-item theme--light">
    <div class="v-list-item__content pa-0 ma-0">
      <div flat="" class="align-center text-center py-4"></div>
    </div>
  </div>
  <div tabindex="-1" role="listitem" class="text-center v-list-item theme--light">
    <div class="v-list-item__content py-1">
      <div class="v-list-item__title font-weight-bold title"></div>
      <p class="body-2 grey--text"></p>
      <p class="caption font-weight-bold my-2 align-bottom"></p>
    </div>
  </div>
</div>
`;
